build.proj

```diff
@@ -39,7 +43,8 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="BuildNative">
+  <!-- We don't have any native components when building for netfx. -->
+  <Target Name="BuildNative" Condition="'$(TargetGroup)' != 'netfx'">
```

Comment:
It seems a bit unusual for this information to be recorded in the root build.proj, instead I would have expected out native build scripts to be somewhat aware of the configuration we're building for and condition things appropriately.

Reply:
Thanks, fixed.

Reply:
You mean UAP binaries in netcoreapp? I thought you meant the aot components in build-native. Did you in-fact mean UAP ref/src binaries are being built in the netcoreapp run? Can you point me to a log where this is happening? I'm happy to fix this as well as part of another PR.

Reply:
Yes, AOT components.  Also AllConfigurations is building native components but I don't believe we ever use them.  

Ultimately it feels a bit hacky that our native build phase is largely oblivious to the configuration of the build which feels to me like we'll always be in a place where we're overbuilding since we don't have a well established pattern.  That said we haven't needed it too much so perhaps its not the biggest issue.

Reply:
And yes, during the native build the configuration isn't respected but pre-conditions work fine as you can just suppress it to run as I did here for netfx.

Reply:
AOT is part of the targetgroup and something we pull out in the `TargetsAot` property.  I actually think it's a misnomer today that we call this the AOT build.  The difference between AOT and non-AOT is the `appcontainer` [bit on the binary](https://github.com/dotnet/corefx/blob/23521cc75428e38b8ff62cbac188c86b32184d42/src/Native/build-native.cmd#L158-L159) which has nothing to do with the "aot"-ness of the configuration and everything to do with running in appcontainer which is part of the UAP framework.  If I had to guess I would imagine that its busted if we happen to use those binaries for netcoreappaot today.

Reply:
That sounds right to me.